core/lib/dal/src/protocol_versions_dal.rs

```diff
@@ -98,6 +98,45 @@ impl ProtocolVersionsDal<'_, '_> {
         db_transaction.commit().await.unwrap();
     }
 
+    pub async fn save_genesis_upgrade(&mut self, id: ProtocolVersionId, tx_hash: Option<H256>) {
+        sqlx::query!(
+            r#"
+            UPDATE protocol_versions
+            SET
+                upgrade_tx_hash = $1
+            WHERE
+                id = $2
+            "#,
+            tx_hash.map(|tx_hash| tx_hash.0.to_vec()),
+            id as i32,
+        )
+        .execute(self.storage.conn())
+        .await
+        .unwrap();
+    }
+
+    pub async fn save_genesis_upgrade_with_tx(
```

Comment:
let's add a comment (and mention that it also injects the transaction in the database)